### PR TITLE
allowed ReadFile() timeout and added handling

### PR DESCRIPTION
**Pull request by ericjtaylor**
will open up some options to address #2

### CODE REVIEW

File: src/echo_chan.cpp [hunk 1]
```diff
@@ -98,39 +88,43 @@ double EchoChan::getSpinnerRotation(int snum) {
 }
 
 void EchoChan::extractStateFromPins() {
-  char buf[11] = {0};
-  char byte[1] = {0};
+  char buf[11] = { 0 };
+  char byte[1] = { 0 };
+  long unsigned int n_read = 0;
```
Comment (HWXLR8) [inline src/echo_chan.cpp:93]: why `long unsigned int` ? surely a regular `unsigned int` is long enough for the number of bytes read? also remember my comments before about descriptive variable names. something like "bytes_read" would require very little effort to understand compared to `n_read`.

Comment (ericjtaylor) [reply src/echo_chan.cpp:93]: ReadFile() in the Windows API wants to receive an LPDWORD, eg. the address of a DWORD which is a 32 bit unsigned integer. Not sure the best practice here, I didn't want to go all in on using the language of Windows types but I still need to respect their function specifications.

Comment (ericjtaylor) [reply src/echo_chan.cpp:93]: not sure the best naming convention to follow. imo the `n_` prefix is handy at communicating something will be a counter. if it's not too wordy how does `n_bytes_read` sound to you?

Comment (HWXLR8) [reply src/echo_chan.cpp:93]: > `DWORD` which is a 32 bit unsigned integer.

an standard `unsigned int` on pretty much all modern platforms is 32 bits, `long` would make it 64 bits. if you want to be 100% sure of size, use `uint32_t` provided by `stdint` . 

`n_bytes_read` sounds fine.

**Comment by ericjtaylor** [comment]
PR is revised per comments now

**Comment by HWXLR8** [comment]
thank you
